### ISSUE TITLE
Provide more stats on each platform message

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#108352]
When optimizing for latency, it is helpful to identify *interesting* platform messages. These can be, but is not limited to:

- Platform messages that send a large amount of data
- Platform messages that take a long time in the method codecs during encoding and decoding
- Platform messages that wait a long time when jumping from the Dart UI thread to the host task queue (for that platform channel)
- Platform messages that wait a long time when jumping from the host task queue to the Dart UI thread
- Platform messages handlers (e.g. `onMethodCall`) that take a long time to run

This can guide performance work by developers. For example, if a platform message takes a long time to jump from the Dart UI thread to the host task queue, such as in instances where the platform thread servicing the default host task queue is blocked, one option would be to use a background task queue for that channel.

Related is https://github.com/flutter/flutter/pull/104531 which provides aggregated stats on the platform messages, but we would like similar data for *each* platform message.

cl/463046259 is a hack that does this, by passing down an identifier from Dart to the engine and host to track the platform message invocation, and then using a perfetto query to parse the timeline events. There are probably better ways to do this though, and it would be nice if this can be eventually integrated into devtools in some sort of table.

cc @gaaclarke @kenzieschmoll

**Comment by gaaclarke** [comment]
I think the way you have the data laid out per platform channel message is clear and nice.  I'm not convinced yet that it is worth the runtime cost and engineering cost.

> Platform messages that send a large amount of data

The aggregate data we already have can give you this information per channel.  I'm not sure it is helpful to the developer to be able to pinpoint specifics via an enumeration that they didn't specify.

> Platform messages that take a long time in the method codecs during encoding and decoding

This is mostly just a function of the payload size.  CPU profilers can give us this information but not tie it back directly to a channel.

> Platform messages that wait a long time when jumping from the Dart UI thread to the host task queue (for that platform channel)
> Platform messages that wait a long time when jumping from the host task queue to the Dart UI thread

We have timeline events for that now.

> Platform messages handlers (e.g. onMethodCall) that take a long time to run
 
CPU profilers can tell us that.  Maybe you are making the case that we shouldn't have to rely on CPU profilers to get this information?

**Comment by jiahaog** [comment]
I agree as well that it may not be worth the cost of building this, let's look at the data first in the prototype and see what action items we can find before we invest more into building a full-fledged solution.
